ledger/streaming/client/src/backfill.rs

```diff
@@ -1,59 +1,59 @@
 // Copyright (c) 2018-2022 The MobileCoin Foundation
 
-//! A BlockStream that backfills missing indices from a given BlockFetcher.
+//! A [BlockStream] that backfills another [BlockStream] using a [BlockFetcher].
 
 use displaydoc::Display;
 use futures::{FutureExt, Stream, StreamExt};
 use mc_common::logger::{log, Logger};
 use mc_ledger_streaming_api::{BlockFetcher, BlockStream, BlockStreamComponents, Result};
 use mc_transaction_core::BlockIndex;
-use std::{pin::Pin, sync::Arc};
+use std::pin::Pin;
 
 /// A [BlockStream] that backfills another [BlockStream] using a [BlockFetcher].
 #[derive(Debug, Display)]
-pub struct BackfillingStream<S: BlockStream + 'static, F: BlockFetcher + 'static> {
+pub struct BackfillingStream<S: BlockStream, F: BlockFetcher> {
     upstream: S,
-    fetcher: Arc<F>,
+    fetcher: F,
     logger: Logger,
 }
 
-impl<S: BlockStream + 'static, F: BlockFetcher + 'static> BackfillingStream<S, F> {
+impl<S: BlockStream, F: BlockFetcher> BackfillingStream<S, F> {
     /// Instantiate a [BackfillingStream].
     pub fn new(upstream: S, fetcher: F, logger: Logger) -> Self {
         Self {
             upstream,
-            fetcher: Arc::new(fetcher),
+            fetcher,
             logger,
         }
     }
 }
 
-impl<S: BlockStream + 'static, F: BlockFetcher + 'static> BlockStream for BackfillingStream<S, F> {
-    type Stream = impl Stream<Item = Result<BlockStreamComponents>>;
+impl<S: BlockStream, F: BlockFetcher> BlockStream for BackfillingStream<S, F> {
+    type Stream<'s>
+    where
+        Self: 's,
+    = impl Stream<Item = Result<BlockStreamComponents>> + 's;
 
-    fn get_block_stream(&self, starting_height: u64) -> Result<Self::Stream> {
+    fn get_block_stream(&self, starting_height: u64) -> Result<Self::Stream<'_>> {
         self.upstream
             .get_block_stream(starting_height)
             .map(|upstream| {
                 backfill_stream(
                     upstream,
                     starting_height,
-                    self.fetcher.clone(),
+                    &self.fetcher,
                     self.logger.clone(),
                 )
             })
     }
 }
 
-fn backfill_stream<
-    S: Stream<Item = Result<BlockStreamComponents>> + 'static,
-    F: BlockFetcher + 'static,
->(
+fn backfill_stream<'f, S: Stream<Item = Result<BlockStreamComponents>> + 'f, F: BlockFetcher>(
```

Comment:
Can some comments be added to the function + function body to annotate what's going on?